source/KeyHandlers.js

```diff
@@ -60,12 +60,30 @@ var onKey = function ( event ) {
     if ( this._keyHandlers[ key ] ) {
         this._keyHandlers[ key ]( this, event, range );
     } else if ( key.length === 1 && !range.collapsed ) {
+        var initialStartOffset = range.startOffset
         // Record undo checkpoint.
         this.saveUndoState( range );
         // Delete the selection
         deleteContentsOfRange( range, this._root );
         this._ensureBottomLine();
-        this.setSelection( range );
+
+        // double clicking a word and typing deletes the space after it
+        // caused because a '  ' is automatically cleaned by the HTML to ' '
+        // the fix inserts a zero width space in between the spaces so it looks like => ' \u200b '
+        // and leaves it selected so it is replaced by the onKey event
+
+        // if the delete was successful, the startContainer should be the endContainer
+        if (range.startContainer === range.endContainer) {
```

Comment:
@rahulgs12 I think we should also check for the two spaces (one before the cursor, and one after), and only use the fix in that case.